### PR TITLE
Fix partner proxy bypass

### PR DESCRIPTION
**Pull request by DuncanUszkay1**
### WHY are these changes introduced?

When running through the end to end guide for the Scripts Platform on fresh installs of every service, I didn't find a way to get a script installed onto `shop1` without database manipulation or going through the process of installing an app.

Ideally we don't need to rely on Partners for our E2E

### WHAT is this pull request doing?

We used to use a flag `BYPASS_PARTNERS_PROXY` which allowed us to call Script Service directly when pushing and enabling Scripts. Somewhere along the line, this flag broke and now we always send a `0` as our `shop_id` due to this line `shop_id: shop_domain&.to_i`. I'm updating that line to send a 1, rather than trying to convert the shop domain string into an integer.

This should only effect local development. If there's a purpose to this line that I missed please let me know

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows).

### CODE REVIEW

**Review by jacobsteves — APPROVED** [review]

File: lib/project_types/script/layers/infrastructure/script_service.rb [hunk 1]
```diff
@@ -117,7 +117,7 @@ def self.api_client(ctx, api_key, shop_domain)
               url: "https://script-service.myshopify.io/graphql",
               token: "",
               api_key: api_key,
-              shop_id: shop_domain&.to_i
```
Comment (andyw8) [inline lib/project_types/script/layers/infrastructure/script_service.rb:120]: @jacobsteves do you recall why it was done this way in https://github.com/Shopify/shopify-app-cli/pull/584 ?

Comment (jacobsteves) [reply lib/project_types/script/layers/infrastructure/script_service.rb:120]: #584 copied over the `deploy` command from script's private fork of the CLI. [Heres](https://github.com/Shopify/temp-shopify-app-cli/pull/181#issuecomment-607398150) the original PR for reference. 

TLDR, we wanted a way to deploy to our service during development without having to go through Partners, and our service only knows about `shop_id` and not `shop_domain`. The plan was to set the `shop_domain` to the shop's ID whenever we wanted to push directly to the service. This approach let us do what we want without cluttering the CLI with development convenience code

Comment (DuncanUszkay1) [reply lib/project_types/script/layers/infrastructure/script_service.rb:120]: 🤔 I wonder if we can make that more explicit. We could change up the env variable to allow for the expression of the Shop ID. `export BYPASS_PARTNER_PROXY_SHOP_ID = <shop ID>`

**Comment by DuncanUszkay1** [comment]
I've made a change to support setting the ID in the env, so that we don't need to make a code change in the future if we need to publish onto a different shop.

**Review by jacobsteves — APPROVED** [review]
